yarn-project/acir-simulator/src/acvm/acvm.ts

```diff
@@ -100,7 +171,25 @@ export async function acvm(
         throw err;
       }
     },
-  );
+  ).catch(err => {
+    // ACVM_js throws raw string errors
+    if (typeof err !== 'string') {
+      throw err;
+    }
+
+    const opcodeLocation = extractOpcodeLocationFromError(err);
+    if (!opcodeLocation || !debug) {
+      throw err;
+    }
+
+    const callStack = getCallStackFromOpcodeLocation(opcodeLocation, debug);
+    logger(printErrorStack(callStack));
+
+    // The ACVM only lets string errors pass through so we need to throw a string at the execution level.
+    // We should probably update the ACVM to let proper errors through.
+    throw `Assertion failed: '${callStack.pop()?.assertionText ?? 'Unknown'}'`;
```

Comment:
But the first `if` throws the error when it's not a string. Should we convert that to a string or is it actually ok to throw `new Error(str)`?

Reply:
The problem is that the acvm js only lets error strings through, so if we emitted new Error(str), if the execution of this ACVM was child of another ACVM, when passing through it'd get transformed to Error in oracle call: unknown or something like that ): 

Maybe we can ignore the ACVM error completely if there was an error in a nested call...

Reply:
I might try that so we have proper errors from the simulator executions